lib/bigtable/filter.js

```diff
@@ -0,0 +1,896 @@
+ * attempting to match an arbitrary byte, you should therefore use the escape
+ * sequence '\C', which may need to be further escaped as '\\C' in your client
+ * language.
+ *
+ * Transformers alter the input row by changing the values of some of its
+ * cells in the output, without excluding them completely. Currently, the only
+ * supported transformer is the {module:bigtable/filter#value} `strip` filter,
+ * which replaces every cell's value with the empty string.
+ *
+ * The total serialized size of a filter message must not
+ * exceed 4096 bytes, and filters may not be nested within each other to a depth
+ * of more than 20.
+ *
+ * Use the following table for the various examples found throughout the
+ * filter documentation.
+ *
+ * | Row Key     | follows:gwashington | follows:jadams | follows:tjefferson |
+ * | ----------- |:-------------------:|:--------------:|:------------------:|
+ * | gwashington |                     | 1              |                    |
+ * | tjefferson  | 1                   | 1              |                    |
+ * | jadams      | 1                   |                | 1                  |
+ *
+ * @constructor
+ * @alias module:bigtable/filter
+ */
+function Filter() {
+  this.filters_ = [];
+}
+
+/**
+ * @private
+ * @throws TypeError
+ *
+ * Transforms Arrays into a simple regular expression for matching multiple
+ * values.
+ *
+ * @param {regex|string|string[]} regex - Either a plain regex, a regex in
+ *     string form or an array of strings.
+ *
+ * @return {string}
+ *
+ * @example
+ * var regexString = Filter.convertToRegExpString(['a', 'b', 'c']);
+ * // => '(a|b|c)'
+ */
+Filter.convertToRegExpString = function(regex) {
+  if (is.regexp(regex)) {
+    return regex.toString().replace(/^\/|\/$/g, '');
+  }
+
+  if (is.array(regex)) {
+    return '(' + regex.join('|') + ')';
+  }
+
+  if (is.string(regex)) {
+    return regex;
+  }
+
+  if (is.number(regex)) {
+    return regex.toString();
+  }
+
+  throw new TypeError('Can\'t convert to RegExp String from unknown type');
+};
+
+/**
+ * @private
+ *
+ * Creates a range object. All bounds default to inclusive.
+ *
+ * @param {?object|string} start - Lower bound value.
+ * @param {?object|string} end - Upper bound value.
+ * @param {string} key - Key used to create range value keys.
+ *
+ * @return {object}
+ *
+ * @example
+ * var range = Filter.createRange('value1', 'value2', 'Test');
+ * // {
+ * //   startTestInclusive: 'value1', // value would be buffer
+ * //   endTestExclusive: 'value2' // value would be buffer
+ * // }
+ *
+ * //-
+ * // It's also possible to pass in objects to specify inclusive/exclusive
+ * // bounds.
+ * //-
+ * var upperBound = {
+ *   value: 'value3',
+ *   inclusive: false
+ * };
+ *
+ * var range = Filter.createRange(upperBound, null, 'Test2');
+ * // => {
+ * //   startTest2Exclusive: 'value3'
+ * // }
+ */
+Filter.createRange = function(start, end, key) {
```

Comment:
I think that's the version we're using, I included it here for completeness, but we also accept a `keys` option within `Table#getRows`.